lib/addressable/template.rb

```diff
@@ -896,19 +896,16 @@ def join_values(operator, return_value)
     #
     # @return [Hash, Array, String] The normalized values
     def normalize_value(value)
-      unless value.is_a?(Hash)
-        value = value.respond_to?(:to_ary) ? value.to_ary : value.to_str
-      end
-
       # Handle unicode normalization
-      if value.kind_of?(Array)
-        value.map! { |val| normalize_value(val) }
+      if value.respond_to?(:to_ary)
+        value.to_ary.map! { |val| normalize_value(val) }
```

Comment:
Here I just simplified a bit the code, treating all array-like object in the same place instead of having the first like (900) to convert first.